hadoop-mini-clusters-hbase/src/main/java/com/github/sakserv/minicluster/impl/HbaseRestLocalCluster.java

```diff
@@ -0,0 +1,363 @@
+            this.hbaseRestPort = hbaseRestPort;
+            return this;
+        }
+
+        public Builder setHbaseRestInfoPort(Integer hbaseRestInfoPort) {
+            this.hbaseRestInfoPort = hbaseRestInfoPort;
+            return this;
+        }
+
+        public Builder setHbaseRestHost(String hbaseRestHost) {
+            this.hbaseRestHost = hbaseRestHost;
+            return this;
+        }
+
+        public Builder setHbaseRestReadOnly(Boolean hbaseRestReadOnly) {
+            this.hbaseRestReadOnly = hbaseRestReadOnly;
+            return this;
+        }
+
+        public Builder setHbaseRestThreadMin(Integer hbaseRestThreadMin) {
+            this.hbaseRestThreadMin = hbaseRestThreadMin;
+            return this;
+        }
+
+        public Builder setHbaseRestThreadMax(Integer hbaseRestThreadMax) {
+            this.hbaseRestThreadMax = hbaseRestThreadMax;
+            return this;
+        }
+
+        public HbaseRestLocalCluster build() {
+            HbaseRestLocalCluster hbaseRestLocalCluster = new HbaseRestLocalCluster(this);
+            validateObject(hbaseRestLocalCluster);
+            return hbaseRestLocalCluster;
+        }
+
+        public void validateObject(HbaseRestLocalCluster hbaseRestLocalCluster) {
+            if (hbaseRestLocalCluster.hbaseMasterPort == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: HBase Master Port");
+            }
+            if (hbaseRestLocalCluster.hbaseMasterInfoPort == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: HBase Master Info Port");
+            }
+            if (hbaseRestLocalCluster.numRegionServers == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: HBase Number of Region Servers");
+            }
+            if (hbaseRestLocalCluster.hbaseRootDir == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: HBase Root Dir");
+            }
+            if (hbaseRestLocalCluster.zookeeperPort == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: Zookeeper Port");
+            }
+            if (hbaseRestLocalCluster.zookeeperConnectionString == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: Zookeeper Connection String");
+            }
+            if (hbaseRestLocalCluster.zookeeperZnodeParent == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: Zookeeper Znode Parent");
+            }
+
+            if (hbaseRestLocalCluster.hbaseRestPort == null) {
+                throw new IllegalArgumentException("ERROR: Missing required config: HBase Rest Port");
+            }
+        }
+    }
+
+    @Override
+    public void start() throws Exception {
+        VersionInfo.logVersion();
+        Configuration conf = HBaseConfiguration.create();
```

Comment:
Can be dangerous I think, I will load all the *-site.xml and *-core.xml by default from the classpath. Maybe a boolean "default load from classpath"?